### ISSUE TITLE
Stellargraph model does not improve over multiple epochs

### ISSUE THREAD

**Issue by ocesp98** [stellargraph/stellargraph#727]
When I am using stellargraph and I fit the generator to my training data, the accuracy and the loss remain the same over all the epochs. I always get this warning when I run following code: 

`history = model.fit_generator(
    train_gen,
    epochs=50,
    validation_data=val_gen,
    verbose=2,
    shuffle=False,  
    callbacks=[es_callback, mc_callback],
)`

> WARNING:tensorflow:From C:\Users\X\Anaconda3\lib\site-packages\tensorflow\python\ops\math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where

I don't know if the warning has anything to do with this. I played around with all parameters and I used a weighted loss function and I also switched weights.

**Comment by huonw** [comment]
Hi @ocesp98, just like on https://github.com/stellargraph/stellargraph/issues/712, we're very happy to help, but we need more information. The template that appears when you click "New Issues > Bug Report (Get Started)" is a great place to start with the information we need:

![image](https://user-images.githubusercontent.com/1203825/73234261-594cba80-41dd-11ea-8a27-76ba68cb0d57.png)

To get us started:

- what version of stellargraph are you using?
- could you include more code? We'll need to see how you create `train_gen` and `val_gen` (for example, we cannot even tell what algorithm/models you are using), as well as how you create the graph itself.

**Comment by ocesp98** [comment]
I'm using Stellargraph version 0.8.4; Python 3.6; NetworkX 2.3. 
Given that I followed the example code when building the stellargraph and given I don't get an error code, I assume that the problem occurs during the preprocessing step. 

I built the network on 1 node type (transaction) and I connected the transaction nodes to eachother when they had the same merchant/client ID. 
I used the undirected GraphSAGE method. 

````
G= nx.Graph()

G.add_nodes_from(target_data.transid)
edgelist = []
for i  in range(len(target_data.client)):
    for j in range(len(target_data.client)):
        if target_data.client.iloc[i] == target_data.client.iloc[j]:
            edgelist.append([target_data.transid.iloc[i], target_data.transid.iloc[j]])

            
print(edgelist)
edgelist2 = []
for i  in range(len(target_data.merchant)):
    for j in range(len(target_data.merchant)):
        if target_data.merchant.iloc[i] == target_data.merchant.iloc[j]:
            edgelist2.append([target_data.transid.iloc[i], target_data.transid.iloc[j]])

````
Then I added the edges from the two edgelists while removing the selfloop_edges
Next, I made all features numeric 

`train_gen = generator.flow(train_data.transid, train_targets, shuffle=True)`

The node features I use only have 5 columns. When looking at the example code, I noticed that the number of columns is a lot higher. Could this be a problem?

The weird thing is that it predicts only the minority class (of which only 2 instances were fed to the train generator). The most logical problem would be that it'd predict only the majority class, however, in my case, it predicts the minority class.

Another difference between my code and the example code is that my node features can be large numerical values, whereas in the example code, it is mainly between 0 and 1. However, it remains weird that the model does not improve and predicts the minority class.
